Documentation for Adaptive Overload Protection (#26690)
hashicorp/vault: website/content/docs/configuration/adaptive-overload-protection.mdx

---
layout: docs
page_title: Adaptive Overload Protection - Configuration
description: |-
  Vault Enterprise provides Adaptive Overload Protection to automatically
  prevent workloads from overloading different resources of the Vault servers.
---

# `adaptive_overload_protection`

@include 'alerts/enterprise-only.mdx'

@include 'alerts/beta.mdx'

> The `adaptive_overload_protection` stanza allows operators to control overload
> protection features. Changes to this stanza require a server restart to take
> effect.

Reviewer: ```suggestion
Configure the `adaptive_overload_protection` stanza to control the overload
protection features for your Vault server.

<Warning title="Restart required">

You must restart your Vault server for the stanza updates to take effect.

</Warning>
```

Reviewer: We should probably make the restart warning a partial so we can include in elsewhere.

For more information read [Adaptive Overload
Protection](/vault/docs/concepts/adaptive-overload-protection).

```hcl
adaptive_overload_protection {
  disable_write_controller = false
}
```

## `adaptive_overload_protection` parameters

> These parameters apply to the `adaptive_overload_protection` stanza in the Vault
> configuration file:

Reviewer: ```suggestion
```
It's obvious that the parameters are for `adaptive_overload_protection` given the section heading, the page title, and the doc intro. No need to repeat the introduction here.

> - `disable_write_controller` `(bool: <optional>)`: Disables the adaptive write
>   overload controller. Defaults to `true` (controller disabled). Set this to
>   `false` explicitly to enable the write controller during the beta period.

Reviewer: ```suggestion
- `disable_write_controller` `(bool: <optional>)`: Disables the adaptive write
  overload controller. The controller is disabled by default
  (`disable_write_controller` = `true`). Set `disable_write_controller` to
  `false` to enable the write controller and opt-in to the beta functionality.
```

Style correction: avoid "this" as a pronoun

> ~> **Warning** This feature is still in Beta. We recommend testing this in
> pre-production with realistic workloads and traffic levels.

Reviewer: ```suggestion
```
We're already including the beta notice at the top of the page so this is redundant. If we want to call out a testing recommendation, it's probably better to make it a tip independent of the beta status
